### PR TITLE
Add overview of all rules to the documentation

### PR DESCRIPTION
**Pull request by fatbasstard**
Added/generated al list of all the rules present in the toolkit. Cleaned up the Readme a bit by moving some of the extended documentation to a separate `docs` folder

Also added an `--update-documentation` to generate the rules documentation (Kept the argument a bit more "general" in case there is other documentation we can generate in the future. 
Hiding it from the help (`-h`).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by kddejong** [comment]
It would be awesome to have a way to update these rule docs automatically... may need to think about that going forward.

**Comment by fatbasstard** [comment]
I've got a generation script in the codebase on my pc... It has some identical behaviour like the updating of the specification. It's not really related to linter, but more "Maintenance" on it  (Run a command and make a PR of the outcome :P)

I'll let that sink for a moment 🤔

File: README.md [hunk 1]
```diff
@@ -43,8 +43,7 @@ There are IDE plugins available to get direct linter feedback from you favorite
 | --append-rules | append_rules | [RULESDIR [RULESDIR ...]] | Specify one or more rules directories using one or more --append-rules arguments. |
 | --ignore-checks | ignore_checks | [IGNORE_CHECKS [IGNORE_CHECKS ...]] | Only check rules whose ID do not match or prefix these values.  Examples: <br />- A value of `W` will disable all warnings<br />- `W2` disables all Warnings for Parameter rules.<br />- `W2001` will disable rule `W2001` |
 | --log-level | log_level | {info, debug} | Log Level |
-| --update-specs | | | Update the CloudFormation Specs.  You may need sudo to run this.  You will need internet access when running this command |
```
Comment (cmmeyer) [inline README.md]: I disagree with hiding/suppressing the update options.  Being able to manually update the resource spec is hugely useful for folks who want to quickly adopt a new service. Some folks want access the minute they are available.

File: src/cfnlint/core.py [hunk 2]
```diff
@@ -176,7 +177,11 @@ def append_parser(parser, defaults):
         version='%(prog)s {version}'.format(version=__version__)
     )
     parser.add_argument(
-        '--update-specs', help='Update the CloudFormation Specs',
+        '--update-specs', help=argparse.SUPPRESS,
```
Comment (cmmeyer) [inline src/cfnlint/core.py]: Same here. Why suppress? Give the folks the option to update if they add local rules or want the latest spec.

Comment (fatbasstard) [reply src/cfnlint/core.py]: That still works, just "hidden it", put it back to what it was?

**Pull request merged**